sling/pyapi/pystore.cc

```diff
@@ -260,11 +260,17 @@ PyObject *PyStore::Symbols() {
 PyObject *PyStore::NewFrame(PyObject *arg) {
   // Check that store is writable.
   if (!Writable()) return nullptr;
-
-  // Parse data into slot list.
+  
   GCLock lock(store);
   std::vector<Slot> slots;
-  if (!SlotList(arg, &slots)) return nullptr;
+
+  // If the argument is a string, create a frame with that id.
+  if (PyString_Check(arg)) {
+    slots.emplace_back(Handle::id(), SymbolValue(arg));
```

Comment:
What about just using store['myid'] to create an empty frame? Then you don't need any special support in NewFrame. This will also take care of the proxy flag.

Reply:
As discussed earlier, store["myid"] creates an unbound symbol that is not saved, e.g. during Encode().